File: tools/src/main/java/com/hardware/tools/configuration/ConfigProperties.java
package com.hardware.tools.configuration;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

public class ConfigProperties {

    private static ConfigProperties instance;
    private static final Properties properties = new Properties();

//    private static final String properties_path = "config.properties";
    private static final String properties_path = "D:\\Projects\\Mini\\WEB\\hardware_store_back\\config.properties";

    private ConfigProperties() throws IOException {
        FileInputStream inputStream = new FileInputStream(ConfigProperties.properties_path);
        properties.load(inputStream);
    }

    public static ConfigProperties getInstance() throws IOException {
        if(instance == null) {
            instance = new ConfigProperties();
        }
        return instance;
    }

    public String getProperty(String key)  {
        return properties.getProperty(key);
    }
}

File: auth/src/main/kotlin/com/hardware/auth/configuration/ConfigProperties.kt
package com.hardware.auth.configuration

import java.util.Properties
import java.io.FileInputStream
import java.lang.NullPointerException

/**
 * Singleton object for loading configuration properties from a file.
 */
object ConfigProperties {
    private val properties = Properties()
    private const val properties_path = "D:\\Projects\\Mini\\WEB\\hardware_store_back\\config.properties"

    /**
     * Initializes the [properties] object with the properties file at [properties_path].
     */
    init {
        val inputStream = FileInputStream(properties_path)
        properties.load(inputStream)
    }

    /**
     * Returns the value of the property with the given [key].
     * @param key the key of the property to retrieve as a [String].
     * @return the value of the property as a [String].
     */

    fun getProperty(key: String): String {
        return properties.getProperty(key) ?: throw NullPointerException("Property $key not found")
    }
}
